receiver/k8sobjectsreceiver/config.go

```diff
@@ -96,6 +98,14 @@ func (c *Config) Validate() error {
 			object.Interval = defaultPullInterval
 		}
 
+		if object.Mode == PullMode && object.ResourceVersion != "" {
+			return fmt.Errorf("resource version is invalid for mode: %v", object.Mode)
```

Comment:
Can we apply this config for the pull mode as well? If provided, we pull resources only for that particular version. If you can add it in a separate PR, that would be great.

Reply:
For the watch mode, `resourceVersion` is used to help with watch consistency and ensure clients don't miss any updates. 

Consistency is not required for the pull mode as it always pulls the latest list of objects.
Although, using `resourceVersion` along with `resourceVersionMatch` can improve the efficiency of the API server.
Quoting the documentation:

**_Unless you have strong consistency requirements, using resourceVersionMatch=NotOlderThan and a known resourceVersion is preferable since it can achieve better performance and scalability of your cluster than leaving resourceVersion and resourceVersionMatch unset, which requires quorum read to be served._**


https://kubernetes.io/docs/reference/using-api/api-concepts/#resource-versions

Reply:
I meant the configuration interface consistency. As a user, I would expect `resource_version` to be applied to both modes even

Reply:
I'm preparing a PR with that change.

Reply:
@dmitryax @Harshit221 #18911 done